swarm-cluster-using-docker-extension/azuredeploy.json

```diff
@@ -0,0 +1,396 @@
+			"[concat('Microsoft.Storage/storageAccounts/', parameters('newStorageAccountName'))]",
+			"[concat('Microsoft.Network/networkInterfaces/', variables('nicName'), copyIndex())]"
+		],
+		"properties": {
+			"hardwareProfile": {
+				"vmSize": "[variables('vmSize')]"
+			},
+			"osProfile": {
+				"computername": "[concat(variables('vmName'), copyIndex())]",
+				"adminUsername": "[parameters('adminUsername')]",
+				"adminPassword": "[parameters('adminPassword')]"
+			},
+			"storageProfile": {
+				"imageReference": {
+					"publisher": "[variables('imagePublisher')]",
+					"offer": "[variables('imageOffer')]",
+					"sku" : "[parameters('ubuntuOSVersion')]",
+					"version":"latest"
+				},
+				"osDisk" : {
+					"name": "osdisk",
+					"vhd": {
+						"uri": "[concat('http://', parameters('newStorageAccountName'), '.blob.core.windows.net/', variables('vmStorageAccountContainerName'), '/', variables('OSDiskName'), copyIndex(), '-osdisk.vhd')]"
+					},
+					"caching": "ReadWrite",
+					"createOption": "FromImage"
+				}
+			},
+			"networkProfile": {
+				"networkInterfaces": [
+					{
+						"id": "[resourceId('Microsoft.Network/networkInterfaces',concat(variables('nicName'), copyindex()))]"
+					}
+				]
+			}
+		}
+	},
+	{
+		"type": "Microsoft.Compute/virtualMachines/extensions",
+		"name": "[concat(variables('vmName'), 'master', '/', variables('extensionName'))]",
+		"apiVersion": "2015-05-01-preview",
+		"location": "[variables('location')]",
+		"dependsOn": [
+			"[concat('Microsoft.Compute/virtualMachines/', variables('vmName'), 'master')]"
+		],
+		"properties": {
+			"publisher": "Microsoft.Azure.Extensions",
+			"type": "DockerExtension",
+			"typeHandlerVersion": "1.0",
+			"settings": {
+				"docker": {
+					"port": "2375"
+				},
+				"compose": {
+					"swarm": {
+						"image": "swarm",
+						"restart": "always",
+						"ports": ["5555:2375"],
+						"command": "[concat('manage token://',parameters('swarmClusterId'))]"
+					}
+				}
+			}
+		}
+	},
+	{
+		"type": "Microsoft.Compute/virtualMachines/extensions",
+		"name": "[concat(variables('vmName'), copyIndex(), '/', variables('extensionName'))]",
+		"apiVersion": "2015-05-01-preview",
+		"location": "[variables('location')]",
+		"copy": {
+			"name": "extLoop",
+			"count": "[parameters('slaveCount')]"
+		},
+		"dependsOn": [
+			"[concat('Microsoft.Compute/virtualMachines/', variables('vmName'), copyIndex())]"
+		],
+		"properties": {
+			"publisher": "Microsoft.Azure.Extensions",
+			"type": "DockerExtension",
+			"typeHandlerVersion": "1.0",
+			"settings": {
+				"docker": {
+					"port": "2375"
+				},
+				"compose": {
+					"swarm": {
+						"image": "swarm",
+						"restart": "always",
+						"command": "[concat('join --addr=', reference(concat(variables('nicName'), copyIndex())).ipConfigurations[0].properties.privateIPAddress, ':2375 token://',parameters('swarmClusterId'))]"
+					}
+				}
+			}
+		}
+	}
+    ],
+    "outputs": {
+	"publicIP": {
```

Comment:
output is rather an "endpoint" (proto://ip:port) in this case, so if you output the entire thing you might want to call this key "swarmMasterEndpoint".